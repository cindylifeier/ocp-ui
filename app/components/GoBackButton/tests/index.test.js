import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import 'mock-local-storage';

import { GoBackButton } from '../index';

configure({ adapter: new Adapter() });

describe('<GoBackButton />', () => {
  describe('snapshot tests', () => {
    it('should match snapshot', () => {
      // Arrange
      const label = 'goBackButton';
      const disabled = false;
      const props = {
        label,
        disabled,
      };

      // Act
      const renderedComponent = shallow(<GoBackButton {...props} />);

      // Assert
      expect(renderedComponent).toMatchSnapshot();
    });
  });
});
