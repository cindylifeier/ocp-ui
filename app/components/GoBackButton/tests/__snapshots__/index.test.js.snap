// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GoBackButton /> snapshot tests should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GoBackButton
    disabled={false}
    label="goBackButton"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <StyledFlatButton__StyledFlatButton
        default={true}
        disabled={false}
        fullWidth={true}
        label="goBackButton"
        onClick={[Function]}
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "default": true,
        "disabled": false,
        "fullWidth": true,
        "label": "goBackButton",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <StyledFlatButton__StyledFlatButton
          default={true}
          disabled={false}
          fullWidth={true}
          label="goBackButton"
          onClick={[Function]}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "default": true,
          "disabled": false,
          "fullWidth": true,
          "label": "goBackButton",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
